### PR TITLE
directly remove part from catalog if part don't exists on disk

### PR DESCRIPTION
**Pull request by yuanquan404**
I hereby agree to the terms of the CLA available at: [placeholder]

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
directly remove part from catalog if part don't exists on disk

Detailed description / Documentation draft:
When the user executes the detach partition, part will be moved to the detach directory, and then GCThread will find that part no longer exists on disk when executed. At this time, it will throw an exception and prevent part from being deleted from the catalog

If you are doing this for the first time, it's recommended to read the lightweight [Contributing to ByConity Documentation](https://github.com/ByConity/ByConity/blob/master/CONTRIBUTING.md) guide first.

### CODE REVIEW

**Comment by yuanquan404** [comment]
> hi @yuanquan404 , pls sign the CLA

done.

File: src/CloudServices/CnchPartGCThread.cpp [hunk 1]
```diff
@@ -452,6 +453,13 @@ void CnchPartGCThread::pushToRemovingQueue(
                         cnch_part->remove();
                     remove_parts.emplace_back(cnch_part);
                 }
+                catch (Poco::FileNotFoundException & e)
+                {
+                    /// If the file already has been deleted, we can delete it directly from catalog.
+                    LOG_ERROR(log, "Error occurs when remove part: " + name + " msg: " + e.displayText());
+                    auto cnch_part = (*it)->toCNCHDataPart(storage);
```
Comment (zeromem) [inline src/CloudServices/CnchPartGCThread.cpp:460]: 可以把451行的cnch_part挪到try外面去，这个就不用再转一次了？

Comment (canhld94) [reply src/CloudServices/CnchPartGCThread.cpp:460]: I think this way is also ok, cannot make sure  the previous declared `cnch_part` is at a valid state when exception was thrown.

**Review by canhld94 — APPROVED** [review]

File: src/CloudServices/CnchPartGCThread.cpp (hunk 1 above)
Comment (yuanquan404) [reply src/CloudServices/CnchPartGCThread.cpp:460]: yes

**Pull request merged**